### DISCUSSION TITLE
Lucene search questions

### DISCUSSION THREAD

**Discussion by GurVJ** [mytourbook/mytourbook#1295] [Q&A]
Even after reading Lucene Query Syntax documentation, I am struggling with Lucene search in Tours :
- I did not manage to find tours where the character "-" is used even with the escaping character "\\". Any idea ?
![Capture 001_20240221](https://github.com/mytourbook/mytourbook/assets/64196685/918a28c9-8d6b-49c6-b98a-e4c62cf1aadc)
![Capture 005_20240221](https://github.com/mytourbook/mytourbook/assets/64196685/394ec3f6-f292-4c5f-963a-9689445e2062)

- Some queries return system error messages
![Capture 003_20240221](https://github.com/mytourbook/mytourbook/assets/64196685/31e0fb90-f439-4634-ac81-9b7bcf32e315)

- Where can we find the names of the fields which we can search in (like "title:") ?

**Comment by wolfgang-ch** [comment]
Firstly you have to disable "Use ease searching"

![image](https://github.com/mytourbook/mytourbook/assets/1283445/2ce39f39-c8b4-4f06-b808-5058457ac399)

then searching with keywords are working

![image](https://github.com/mytourbook/mytourbook/assets/1283445/e5821041-bf13-42b1-9208-fe507d4c859c)

![image](https://github.com/mytourbook/mytourbook/assets/1283445/2a068cb8-b05e-409d-9a84-b2599da75ff5)

These are the available search fields

![image](https://github.com/mytourbook/mytourbook/assets/1283445/25423f39-8aaf-4772-a3bc-95ceabbe54fa)

**Comment by GurVJ** [comment]
Thank you @wolfgang-ch for the list of fields.

I always disable "Use ease searching".
I am ok with your example of search, but what about special characters ? Lot of my tour titles have special characters (parenthesis, - , withespaces, ', etc.) like in my example, how can I efficiently search within my tours ?

In your example, how would you search all tours that end with "- Amden" ?
My guess would be title:"\\- Amden" but it doesn't work.

**Reply by wolfgang-ch** [reply]
The fulltext search is a very very complex topic and there a 1 Million or more possibility to tweak the fulltext search. My goal was to provide a fulltext search for MT but not with all possible refinements, this would be it's own job
